src/PostgREST/Request/ApiRequest.hs

```diff
@@ -494,7 +494,11 @@ findProc qi argumentsKeys paramsAsSingleObject allProcs contentType isInvPost =
     [proc] -> Right proc
     procs  -> Left $ AmbiguousRpc (toList procs)
   where
-    matchProc = filter matchesParams $ M.lookupDefault mempty qi allProcs -- first find the proc by name
+    hasSingleUnnamedJsonParam proc = case proc of
+          ProcDescription{pdParams=[ProcParam ppName ppType _ _]} -> T.null ppName && ppType `elem` ["json", "jsonb"]
+          _ -> False
+    matchProc = if null matchProc' && isInvPost then filter hasSingleUnnamedJsonParam (M.lookupDefault mempty qi allProcs) else matchProc'
```

Comment:
> I don't like too much that a second filter is made on the overloaded procedures again though, if it affects performance I'll try to find an alternative.

Addressed that using a `foldr` with a partition to avoid the double filter. It should be ready to review right now.